### PR TITLE
fix: Magic Link Login can be used even if `$allowMagicLinkLogins` is false

### PR DESCRIPTION
**Pull request by datamweb**
Fixed :#777

### CODE REVIEW

**Comment by datamweb** [comment]
IMG_20230809_233657

**Comment by kenjis** [comment]
Re-ran failed jobs.

**Comment by kenjis** [comment]
We have three action methods.
```
$ php spark routes | grep MagicLinkController
| GET    | login/magic-link        | magic-link         | \CodeIgniter\Shield\Controllers\MagicLinkController::loginView     |                | toolbar       |
| GET    | login/verify-magic-link | verify-magic-link  | \CodeIgniter\Shield\Controllers\MagicLinkController::verify        |                | toolbar       |
| POST   | login/magic-link        | »                  | \CodeIgniter\Shield\Controllers\MagicLinkController::loginAction   |                | toolbar       |
```
Can you add the `if` statement to `verify()` just in case?

File: src/Language/ja/Auth.php [hunk 1]
```diff
@@ -51,6 +51,7 @@
     'magicLinkExpired'   => '申し訳ございません、リンクは切れています。', // 'Sorry, link has expired.'
     'checkYourEmail'     => 'メールをチェックしてください！', // 'Check your email!'
     'magicLinkDetails'   => 'ログインリンクが含まれたメールを送信しました。これは {0} 分間だけ有効です。', // 'We just sent you an email with a Login link inside. It is only valid for {0} minutes.'
+    'magicLinkDisabled'  => '(To be translated) Use of MagicLink is currently not allowed.',
```
Comment (kenjis) [inline src/Language/ja/Auth.php:54]:
```suggestion
    'magicLinkDisabled'  => 'マジックリンクは使えません。', // 'Use of MagicLink is currently not allowed.'
```

**Comment by datamweb** [comment]
> Can you add the `if` statement to `verify()` just in case?

Honestly, I thought about this, isn't it better to have a private function?
```php
    private function checkAllowMagicLink( ): RedirectResponse
   { 
        // Check if magic-link is not allowed
        if (!setting('Auth.allowMagicLinkLogins')) {
            return redirect()->route('login')->with('error', lang('Auth.magicLinkDisabled'));
        }
   }
```

**Review by kenjis — APPROVED** [review]

**Comment by kenjis** [comment]
The three `if` statements are the exactly the same and the same meaning.
So it is better not to repeat them.

File: src/Controllers/MagicLinkController.php [hunk 2]
```diff
@@ -50,6 +50,8 @@ public function __construct()
      */
     public function loginView()
     {
+        $this->displayErrorMagicLinkDisabled();
```
Comment (kenjis) [inline src/Controllers/MagicLinkController.php:53]: `RedirectResponse` must be returned from this method if `allowMagicLinkLogins` is false.

**Comment by datamweb** [comment]
@kenjis, method `verify(): RedirectResponse` uses native return type , using a private method causes **breaking change**. Is it worth it?
Or am I missing something?

**Comment by kenjis** [comment]
> using a private method causes breaking change.

No, it is not a breaking change.

But this PR works, so I merge for now.

**Review by kenjis — APPROVED** [review]

**Pull request merged**